news_events/etl/ol_events.py

```diff
@@ -5,171 +5,192 @@
 
     Args:
-        event_data (Soup): The event data BeautifulSoup object
+        image_text_metadata (dict): The image metadata containing alt/title info
+        image_src_metadata(dict): The image metadata containing the image url
 
     Returns:
-        dict: The event image url
-    """
-    image_div = event_data.find("div", class_="field--name-field-event-image")
-    if image_div:
-        img = image_div.find("img")
-        if img:
-            return {
-                "url": urljoin(OL_EVENTS_BASE_URL, img.attrs["src"]),
-                "alt": img.attrs["alt"],
-                "description": img.attrs["alt"],
-            }
+        dict: The transformed event image info
+    """
+    image_src = (
+        image_src_metadata.get("data", {})
+        .get("attributes", {})
+        .get("uri", {})
+        .get("url")
+    )
+    attrs = (
+        image_text_metadata.get("data", {})
+        .get("relationships", {})
+        .get("field_media_image", {})
+        .get("data", {})
+        .get("meta", {})
+    )
+    if image_src:
+        return {
+            "url": urljoin(OL_EVENTS_BASE_URL, image_src),
+            "alt": attrs.get("alt"),
+            "description": attrs.get("title"),
+        }
     return None
 
 
-def transform_event(event_data: Soup, event_page_data: Soup) -> dict:
+def transform_event(event_data: dict) -> dict or None:
     """
     Transform the Open Learning Event data.
 
     Args:
-        event_data (Soup): The event item data BeautifulSoup object
-        event_page_data (Soup): The event page data BeautifulSoup object
+        event_data (dict): The event item data
 
     Returns:
         dict: The transformed event data
     """
+    attributes = event_data.get("attributes", {})
+    event_path = attributes.get("path", {}).get("alias", "")
+    dt = attributes.get("field_event_date", {}).get("value")
+    dt_utc = (
+        parser.parse(dt).replace(tzinfo=ZoneInfo("US/Eastern")).astimezone(UTC)
+        if dt
+        else None
+    )
+    if dt_utc < now_in_utc():
+        return None
+
     return {
-        "guid": event_data.find("a").attrs["href"],
-        "url": urljoin(OL_EVENTS_BASE_URL, event_data.find("a").attrs["href"]),
-        "title": tag_text(event_data.find("h4")),
-        "image": transform_event_image(event_data),
-        "summary": tag_text(
-            event_page_data.find("div", class_="field--type-text-with-summary")
-        ),
-        "content": safe_html(
-            event_page_data.find("div", class_="field--type-text-with-summary")
+        "guid": event_data["id"],
+        "url": urljoin(OL_EVENTS_BASE_URL, event_path) if event_path else None,
+        "title": attributes.get("title"),
+        "image": transform_event_image(
+            *extract_event_image(
+                event_data.get("relationships", {}).get("field_event_image", {})
+            )
         ),
+        "summary": strip_tags(attributes.get("body", {}).get("value") or ""),
```

Comment:
seems like this should be trying to get the summary from the body like so ```attributes.get("body", {}).get("summary")```

Reply:
That's what I was originally thinking of doing but the value is always "".

Reply:
Changed it just in case it might be different than "" in the future, if not the same value will still be in the `content` field